### PR TITLE
[Clang tidy] initialize variables, split declarations

### PR DESCRIPTION
**Pull request by akallabeth**
* cleaned up version of https://github.com/FreeRDP/FreeRDP/pull/9795
* one warning fixed per commit
* based on #9798

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test please

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Review by mfleisz — COMMENTED** [review]
Did a rough view through each single commit - most of it LGTM - just got 2 remarks regarding nullptr dereferencing.

File: libfreerdp/cache/bitmap.c [hunk 1]
```diff
@@ -397,8 +398,7 @@ void bitmap_cache_free(rdpBitmapCache* bitmapCache)
 
 	bitmap_cache_save_persistent(bitmapCache);
 
-	UINT32 i = 0;
-	for (i = 0; i < bitmapCache->maxCells; i++)
+	for (UINT32 i = 0; i < bitmapCache->maxCells && bitmapCache->cells; i++)
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply libfreerdp/cache/bitmap.c]: done

**Review by mfleisz — APPROVED** [review]

**Pull request merged**